src/pages/Encounters/tabs/responses.tsx

```diff
@@ -0,0 +1,223 @@
+import { formatDateTime, formatName } from "@/Utils/utils";
+import { ArrowRight, Plus, X } from "lucide-react";
+import { useQueryParams } from "raviger";
+import { useCallback } from "react";
+import { useTranslation } from "react-i18next";
+import { QuestionnaireDetail } from "./../../../types/questionnaire/questionnaire";
+
+interface ResponsesCardProps {
+  response: QuestionnaireResponse;
+  isActive: boolean;
+  onClick: () => void;
+  showTitle?: boolean;
+}
+function ResponsesCard({
+  response,
+  isActive,
+  onClick,
+  showTitle = true,
+}: ResponsesCardProps) {
+  const { t } = useTranslation();
+  return (
+    <Card
+      onClick={onClick}
+      className={cn(
+        "cursor-pointer p-3 text-sm border hover:border-primary-500 transition-colors",
+        isActive && "border-primary-600 bg-primary-50",
+      )}
+    >
+      <div className="flex items-center justify-between">
+        <div>
+          {showTitle && (
+            <div className="mt-1 font-medium pb-1">
+              {response.questionnaire?.title}
+            </div>
+          )}
+          <div className="font-medium text-xs">
+            {formatDateTime(response.created_date)}
+          </div>
+          <div className="text-gray-600">
+            {t("filed_by")}{" "}
+            <span className="font-medium text-gray-800">
+              {formatName(response.created_by)}
+            </span>
+          </div>
+        </div>
+        {isActive && <ArrowRight className="size-4 text-gray-500" />}
+      </div>
+    </Card>
+  );
+}
+
+export const EncounterResponsesTab = () => {
+  const {
+    selectedEncounter: encounter,
+    patientId,
+    canReadSelectedEncounter: canAccess,
+  } = useEncounter();
+  const { t } = useTranslation();
+  const [qParams, setQueryParams] = useQueryParams<{
+    questionnaireId?: string;
+    questionnaireTitle?: string;
+    responseId?: string;
+  }>();
+
+  const { questionnaireId, responseId } = qParams;
+
+  const selectedQuestionnaire = questionnaireId
+    ? {
+        id: questionnaireId,
+        slug: "",
+        questions: [],
+        title: "",
+        status: "active",
+        subject_type: "encounter",
+        tags: [],
+      }
+    : null;
+
+  const handleQuestionnaireSelect = useCallback(
+    (questionnaire: QuestionnaireDetail) => {
+      setQueryParams({
+        questionnaireId: questionnaire.id,
+      });
+    },
+    [setQueryParams],
+  );
+
+  const handleQuestionnaireClear = useCallback(() => {
+    setQueryParams({});
+  }, [setQueryParams]);
+
+  const handleResponseSelect = useCallback(
+    (responseId: string) => {
+      setQueryParams({
+        ...qParams,
+        responseId,
+      });
+      window.location.hash = `response-${responseId}`;
+    },
+    [setQueryParams, qParams],
+  );
+
+  const handleTitleClick = useCallback(
+    (questionnaireId: string) => {
+      setQueryParams({
+        questionnaireId,
+      });
+    },
+    [setQueryParams],
+  );
```

Comment:
you can omit `useCallback` and directly inline those implementations in react 19 without worrying about memoization